optuna/integration/mlflow.py

```diff
@@ -75,14 +75,29 @@ def objective(trial):
             Name of the metric. Since the metric itself is just a number,
             `metric_name` can be used to give it a name. So you know later
             if it was roc-auc or accuracy.
+        nest_trials:
+            Flag indicating whether or not trials should be logged as
+            nested runs. This is often helpful for aggregating trials
+            to a particular study, under a given experiment.
+        attach_study_attrs:
+            Flag indicating whether or not to attach the study's user attrs
+            to the mlflow trial.
     """
 
-    def __init__(self, tracking_uri: Optional[str] = None, metric_name: str = "value") -> None:
+    def __init__(
+        self,
+        tracking_uri: Optional[str] = None,
+        metric_name: str = "value",
+        nest_trials: Optional[bool] = False,
+        attach_study_attrs: Optional[bool] = False,
```

Comment:
More if a question but would this better communicate the intent (considering `mlflow.set_tags`)?
```suggestion
        tag_study_attrs: Optional[bool] = False,
```

Reply:
I don't have any issue updating to the tag prefix if that makes more sense.

With respect to nesting the runs, setting `nested=True` in the callback's `mlflow.start_run`, causes it to be nested within a parent run, so if you did something like the snippet below, all the Optuna trials would be nested under the 'Parent Run'.

```python
mlfcb = RAPIDSMLflowCallback(...., nest_trials=True)
study = optuna.create_study(....)
with mlflow.start_run(run_name=f"Parent Run"):
    study.optimize(...., callbacks=[mlfcb])
```

Reply:
As for the nesting, makes a lot of sense. Thanks for the code snippet.

The parameter name is also a matter of preference. Unless "attach" is commonly used in mlflow, I'd stick to this suggestion. However, I'm very open to suggestions from especially mlflow users.